flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/SqlFunctionUtils.java

```diff
@@ -1174,6 +1174,107 @@ public static String toBase64(byte[] bytes) {
         return Base64.getEncoder().encodeToString(bytes);
     }
 
+    public static BinaryStringData jsonQuote(BinaryStringData input) {
+        if (input == null) {
+            return null;
+        }
+
+        StringBuilder result = new StringBuilder("\"");
+
+        for (int i = 0; i < input.getSizeInBytes(); ++i) {
+            char c = (char) (input.byteAt(i) & 0xFF);
```

Comment:
I think this doesn't work for multi-byte characters, so it would be better to use code points instead.

Reply:
Good point. Fixed